src/com/reason/ide/files/EsyPackageJsonFile.java

```diff
@@ -0,0 +1,59 @@
+package com.reason.ide.files;
+
+import com.intellij.framework.detection.FileContentPattern;
+import com.intellij.json.JsonFileType;
+import com.intellij.json.psi.JsonFile;
+import com.intellij.json.psi.JsonObject;
+import com.intellij.json.psi.JsonValue;
+import com.intellij.lang.Language;
+import com.intellij.openapi.fileTypes.FileType;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.patterns.ElementPattern;
+import com.intellij.patterns.PatternCondition;
+import com.intellij.psi.FileViewProvider;
+import com.intellij.psi.PsiFile;
+import com.intellij.util.ProcessingContext;
+import com.intellij.util.indexing.FileContent;
+import com.intellij.util.indexing.FileContentImpl;
+import org.jetbrains.annotations.NotNull;
+
+public class EsyPackageJsonFile extends FileBase {
+
+    /* detects any "package.json" with a top-level "esy" property */
+    public static boolean isEsyPackageJson(@NotNull VirtualFile virtualFile) {
```

Comment:
is it just for tests or will it be used after ? If just for tests, maybe we can move that function in tests ?

Reply:
this will be used in the near future. pulled this out of my larger branch